### ISSUE TITLE
Invalid offense/autocorrect for `Style/RedundantLineContinuation`

### ISSUE THREAD

**Issue by MartySixFifty** [rubocop/rubocop#12758]
## Expected behavior

Not to flag the `Style/RedundantLineContinuation` offense.

## Actual behavior

Offense is flagged and incorrectly autocorrected.

### Input:
```
return unless params[:filter].present? && params[:show_inactive].present? \
  && params[:show_inactive] == 'true'
```

### Is flagged as:
```
app/controllers/api/v1/organizations_controller.rb:401:83: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
        return unless params[:filter].present? && params[:show_inactive].present? \ ...
```

### Autocorrects to:
```
return unless params[:filter].present? && params[:show_inactive].present?
  && params[:show_inactive] == 'true'
```

### Then produces:
```
app/controllers/api/v1/organizations_controller.rb:402:11: F: Lint/Syntax: unexpected token tAMPER
(Using Ruby 3.3 parser; configure using TargetRubyVersion parameter, under AllCops)
          && params[:show_inactive] == 'true'
          ^
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.62.0 (using Parser 3.3.0.5, rubocop-ast 1.31.1, running on ruby 3.3.0) [arm64-darwin23]
  - rubocop-rails 2.24.0
```
